compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -1856,13 +1932,28 @@ private void loadTupleType(MethodVisitor mv, BTupleType bType) {
      * @param bType user defined type
      */
     private void loadUserDefinedType(MethodVisitor mv, BType bType) {
+        BTypeSymbol typeSymbol = bType.tsymbol.isTypeParamResolved ? bType.tsymbol.typeParamTSymbol : bType.tsymbol;
+        BType typeToLoad = bType.tsymbol.isTypeParamResolved ? typeSymbol.type : bType;
+        PackageID packageID = typeSymbol.pkgID;
+        String typeOwner = JvmCodeGenUtil.getPackageName(packageID) + MODULE_INIT_CLASS_NAME;
+        String fieldName = getTypeFieldName(toNameString(typeToLoad));
 
-        PackageID packageID = bType.tsymbol.pkgID;
+        if (fieldName.contains(BLangAnonymousModelHelper.ANON_PREFIX)) {
```

Comment:
We only need to do this when loading annon types from other modules right, I think we need to add a check to see if the loaded type is not from the current package being compiled.

Reply:
@warunalakshitha @manuranga Is it possible to get the information about current module being compiled here?

Reply:
We could try adding it as the field to current class (JvmTypeGen) and provide that from the constructor. Have to try it can see.

Reply:
Can we do `&& packageID != PackageID.DEFAULT`?

Reply:
fixed in: 9a611ab